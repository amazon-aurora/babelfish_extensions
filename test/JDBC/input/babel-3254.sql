-- int, bigint, smallint, tinyint, bit
create procedure babel_3254_p1 @a int,
                           @b bigint,
                           @c smallint,
                           @d tinyint,
                           @e bit,
                           @f float,
                           @g real
as select 1
go


-- char, varchar, text, nchar, nvarchar, ntext
create procedure babel_3254_p2 @a char(5),
                           @b nchar(5),
                           @c varchar(5),
                           @d nvarchar(5),
                           @e text,
                           @g ntext,
                           @h varchar(max),
                           @i nvarchar(max),
                           @j char,
                           @k varchar,
                           @l nchar,
                           @m nvarchar
as select 1
go

-- binary, varbinary, image
create procedure babel_3254_p3 @a binary,
                            @b varbinary,
                            @c binary(5),
                            @d varbinary(5),
                            @e image,
                            @f varbinary(max)
as select 1
go

-- decimal and numeric
create procedure babel_3254_p4 @a decimal,
                            @b numeric,
                            @c decimal(5,2),
                            @d numeric(5,2)
as select 1
go

-- date, datetimeoffset, datetime, datetime2, smalldatetime, time
create procedure babel_3254_p5 @a date,
                            @b datetimeoffset,
                            @c datetime,
                            @d datetime2,
                            @e smalldatetime,
                            @f time,
                            @g datetime2(7),
                            @h datetimeoffset(7),
                            @i time(7)
as select 1
go

-- money and smallmoney
create procedure babel_3254_p6 @a money,
                            @b smallmoney
as select 1
go

sp_procedure_params_100_managed 'babel_3254_p1'
go

sp_procedure_params_100_managed 'babel_3254_p2'
go

sp_procedure_params_100_managed 'babel_3254_p3'
go

sp_procedure_params_100_managed 'babel_3254_p4'
go

sp_procedure_params_100_managed 'babel_3254_p5'
go

sp_procedure_params_100_managed 'babel_3254_p6'
go

drop procedure babel_3254_p1
go

drop procedure babel_3254_p2
go

drop procedure babel_3254_p3
go

drop procedure babel_3254_p4
go

drop procedure babel_3254_p5
go

drop procedure babel_3254_p6
go
