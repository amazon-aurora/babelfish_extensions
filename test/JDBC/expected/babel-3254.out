-- int, bigint, smallint, tinyint, bit
create procedure babel_3254_p1 @a int,
                           @b bigint,
                           @c smallint,
                           @d tinyint,
                           @e bit,
                           @f float,
                           @g real
as select 1
go


-- char, varchar, text, nchar, nvarchar, ntext
create procedure babel_3254_p2 @a char(5),
                           @b nchar(5),
                           @c varchar(5),
                           @d nvarchar(5),
                           @e text,
                           @g ntext,
                           @h varchar(max),
                           @i nvarchar(max),
                           @j char,
                           @k varchar,
                           @l nchar,
                           @m nvarchar
as select 1
go

-- binary, varbinary, image
create procedure babel_3254_p3 @a binary,
                            @b varbinary,
                            @c binary(5),
                            @d varbinary(5),
                            @e image,
                            @f varbinary(max)
as select 1
go

-- decimal and numeric
create procedure babel_3254_p4 @a decimal,
                            @b numeric,
                            @c decimal(5,2),
                            @d numeric(5,2)
as select 1
go

-- date, datetimeoffset, datetime, datetime2, smalldatetime, time
create procedure babel_3254_p5 @a date,
                            @b datetimeoffset,
                            @c datetime,
                            @d datetime2,
                            @e smalldatetime,
                            @f time,
                            @g datetime2(7),
                            @h datetimeoffset(7),
                            @i time(7)
as select 1
go

-- money and smallmoney
create procedure babel_3254_p6 @a money,
                            @b smallmoney
as select 1
go

sp_procedure_params_100_managed 'babel_3254_p1'
go
~~START~~
varchar#!#smallint#!#smallint#!#int#!#smallint#!#smallint#!#varchar#!#varchar#!#nvarchar#!#varchar#!#varchar#!#varchar#!#int
@RETURN_VALUE#!#4#!#8#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#int#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@a#!#1#!#8#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#int#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@b#!#1#!#0#!#<NULL>#!#19#!#<NULL>#!#<NULL>#!#<NULL>#!#bigint#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@c#!#1#!#16#!#<NULL>#!#5#!#<NULL>#!#<NULL>#!#<NULL>#!#smallint#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@d#!#1#!#20#!#<NULL>#!#3#!#<NULL>#!#<NULL>#!#<NULL>#!#tinyint#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@e#!#1#!#2#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#bit#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@f#!#1#!#6#!#<NULL>#!#53#!#<NULL>#!#<NULL>#!#<NULL>#!#float#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@g#!#1#!#13#!#<NULL>#!#24#!#<NULL>#!#<NULL>#!#<NULL>#!#real#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
~~END~~


sp_procedure_params_100_managed 'babel_3254_p2'
go
~~START~~
varchar#!#smallint#!#smallint#!#int#!#smallint#!#smallint#!#varchar#!#varchar#!#nvarchar#!#varchar#!#varchar#!#varchar#!#int
@RETURN_VALUE#!#4#!#8#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#int#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@a#!#1#!#3#!#5#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#char#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@b#!#1#!#10#!#5#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#nchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@c#!#1#!#22#!#5#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#varchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@d#!#1#!#12#!#5#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#nvarchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@e#!#1#!#18#!#2147483647#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#text#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@g#!#1#!#11#!#1073741823#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#ntext#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@h#!#1#!#22#!#0#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#varchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@i#!#1#!#12#!#0#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#nvarchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@j#!#1#!#3#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#char#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@k#!#1#!#22#!#0#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#varchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@l#!#1#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#nchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@m#!#1#!#12#!#0#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#nvarchar#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
~~END~~


sp_procedure_params_100_managed 'babel_3254_p3'
go
~~START~~
varchar#!#smallint#!#smallint#!#int#!#smallint#!#smallint#!#varchar#!#varchar#!#nvarchar#!#varchar#!#varchar#!#varchar#!#int
@RETURN_VALUE#!#4#!#8#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#int#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@a#!#1#!#1#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#binary#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@b#!#1#!#21#!#0#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#varbinary#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@c#!#1#!#1#!#5#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#binary#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@d#!#1#!#21#!#5#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#varbinary#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@e#!#1#!#7#!#2147483647#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#image#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@f#!#1#!#21#!#0#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#varbinary#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
~~END~~


sp_procedure_params_100_managed 'babel_3254_p4'
go
~~START~~
varchar#!#smallint#!#smallint#!#int#!#smallint#!#smallint#!#varchar#!#varchar#!#nvarchar#!#varchar#!#varchar#!#varchar#!#int
@RETURN_VALUE#!#4#!#8#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#int#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@a#!#1#!#5#!#<NULL>#!#38#!#38#!#<NULL>#!#<NULL>#!#decimal#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@b#!#1#!#5#!#<NULL>#!#38#!#38#!#<NULL>#!#<NULL>#!#numeric#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@c#!#1#!#5#!#<NULL>#!#38#!#38#!#<NULL>#!#<NULL>#!#decimal#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@d#!#1#!#5#!#<NULL>#!#38#!#38#!#<NULL>#!#<NULL>#!#numeric#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
~~END~~


sp_procedure_params_100_managed 'babel_3254_p5'
go
~~START~~
varchar#!#smallint#!#smallint#!#int#!#smallint#!#smallint#!#varchar#!#varchar#!#nvarchar#!#varchar#!#varchar#!#varchar#!#int
@RETURN_VALUE#!#4#!#8#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#int#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@a#!#1#!#31#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#date#!#<NULL>#!#<NULL>#!#<NULL>#!#0
@b#!#1#!#34#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#datetimeoffset#!#<NULL>#!#<NULL>#!#<NULL>#!#7
@c#!#1#!#4#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#datetime#!#<NULL>#!#<NULL>#!#<NULL>#!#3
@d#!#1#!#33#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#datetime2#!#<NULL>#!#<NULL>#!#<NULL>#!#7
@e#!#1#!#15#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#smalldatetime#!#<NULL>#!#<NULL>#!#<NULL>#!#0
@f#!#1#!#32#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#time#!#<NULL>#!#<NULL>#!#<NULL>#!#7
@g#!#1#!#33#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#datetime2#!#<NULL>#!#<NULL>#!#<NULL>#!#7
@h#!#1#!#34#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#datetimeoffset#!#<NULL>#!#<NULL>#!#<NULL>#!#7
@i#!#1#!#32#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#time#!#<NULL>#!#<NULL>#!#<NULL>#!#7
~~END~~


sp_procedure_params_100_managed 'babel_3254_p6'
go
~~START~~
varchar#!#smallint#!#smallint#!#int#!#smallint#!#smallint#!#varchar#!#varchar#!#nvarchar#!#varchar#!#varchar#!#varchar#!#int
@RETURN_VALUE#!#4#!#8#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#int#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@a#!#1#!#9#!#<NULL>#!#19#!#<NULL>#!#<NULL>#!#<NULL>#!#money#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
@b#!#1#!#17#!#<NULL>#!#10#!#<NULL>#!#<NULL>#!#<NULL>#!#smallmoney#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
~~END~~


drop procedure babel_3254_p1
go

drop procedure babel_3254_p2
go

drop procedure babel_3254_p3
go

drop procedure babel_3254_p4
go

drop procedure babel_3254_p5
go

drop procedure babel_3254_p6
go
